hudi-cli/src/test/java/org/apache/hudi/cli/commands/TestFileSystemViewCommand.java

```diff
@@ -0,0 +1,255 @@
+
+    // Write commit files
+    new File(tablePath + "/.hoodie/" + commitTime1 + ".commit").createNewFile();
+    new File(tablePath + "/.hoodie/" + commitTime2 + ".commit").createNewFile();
+
+    // Reload meta client and create fsView
+    metaClient = HoodieTableMetaClient.reload(metaClient);
+
+    fsView = new HoodieTableFileSystemView(metaClient, metaClient.getActiveTimeline(), true);
+  }
+
+  /**
+   * Test case for 'show fsview all'.
+   */
+  @Test
+  public void testShowCommits() {
+    // Test default show fsview all
+    CommandResult cr = getShell().executeCommand("show fsview all");
+    assertTrue(cr.isSuccess());
+
+    // Get all file groups
+    Stream<HoodieFileGroup> fileGroups = fsView.getAllFileGroups(partitionPath);
+
+    List<Comparable[]> rows = new ArrayList<>();
+    fileGroups.forEach(fg -> fg.getAllFileSlices().forEach(fs -> {
+      int idx = 0;
+      // For base file only Views, do not display any delta-file related columns
+      Comparable[] row = new Comparable[8];
+      row[idx++] = fg.getPartitionPath();
+      row[idx++] = fg.getFileGroupId().getFileId();
+      row[idx++] = fs.getBaseInstantTime();
+      row[idx++] = fs.getBaseFile().isPresent() ? fs.getBaseFile().get().getPath() : "";
+      row[idx++] = fs.getBaseFile().isPresent() ? fs.getBaseFile().get().getFileSize() : -1;
+      row[idx++] = fs.getLogFiles().count();
+      row[idx++] = fs.getLogFiles().mapToLong(HoodieLogFile::getFileSize).sum();
+      row[idx++] = fs.getLogFiles().collect(Collectors.toList()).toString();
+      rows.add(row);
+    }));
+
+    Function<Object, String> converterFunction =
+        entry -> NumericUtils.humanReadableByteCount((Double.parseDouble(entry.toString())));
+    Map<String, Function<Object, String>> fieldNameToConverterMap = new HashMap<>();
+    fieldNameToConverterMap.put(HoodieTableHeaderConfig.HEADER_TOTAL_DELTA_FILE_SIZE, converterFunction);
+    fieldNameToConverterMap.put(HoodieTableHeaderConfig.HEADER_DATA_FILE_SIZE, converterFunction);
+
+    TableHeader header = new TableHeader().addTableHeaderField(HoodieTableHeaderConfig.HEADER_PARTITION)
+        .addTableHeaderField(HoodieTableHeaderConfig.HEADER_FILE_ID)
+        .addTableHeaderField(HoodieTableHeaderConfig.HEADER_BASE_INSTANT)
+        .addTableHeaderField(HoodieTableHeaderConfig.HEADER_DATA_FILE)
+        .addTableHeaderField(HoodieTableHeaderConfig.HEADER_DATA_FILE_SIZE)
+        .addTableHeaderField(HoodieTableHeaderConfig.HEADER_NUM_DELTA_FILES)
+        .addTableHeaderField(HoodieTableHeaderConfig.HEADER_TOTAL_DELTA_FILE_SIZE)
+        .addTableHeaderField(HoodieTableHeaderConfig.HEADER_DELTA_FILES);
+    String expected = HoodiePrintHelper.print(header, fieldNameToConverterMap, "", false, -1, false, rows);
+    assertEquals(expected, cr.getResult().toString());
+
+    // Test command with options, baseFileOnly and maxInstant is 2
+    cr = getShell().executeCommand("show fsview all --baseFileOnly true --maxInstant 2");
```

Comment:
Can we break this command into a new test case? I always suggest one test method verify one thing. Additionally, this method seems a bit long.